### PR TITLE
JS trace filtering fixes

### PR DESCRIPTION
**Pull request by armanbilge**
Some progress/workarounds for #2551. Still not quite right though I think.

### CODE REVIEW

File: core/js/src/main/scala/cats/effect/IOApp.scala [hunk 1]
```diff
@@ -224,7 +224,8 @@ trait IOApp {
               // Do not report cancelation exceptions but still exit with an error code.
               reportExitCode(ExitCode(1))
             case t: Throwable =>
-              throw t
+              t.printStackTrace()
+              reportExitCode(ExitCode(1))
```
Comment (armanbilge) [inline core/js/src/main/scala/cats/effect/IOApp.scala:228]: Throwing an exception on JS causes it to render "natively", which is extremely unhelpful.

File: core/js/src/main/scala/cats/effect/tracing/TracingPlatform.scala [hunk 2]
```diff
@@ -74,14 +74,16 @@ private[tracing] abstract class TracingPlatform { self: Tracing.type =>
...
+    "{anonymous}"
   )
 
   private[this] final val stackTraceFileNameFilter: Array[String] = Array(
     "githubusercontent.com/typelevel/cats-effect/",
     "githubusercontent.com/typelevel/cats/",
     "githubusercontent.com/scala-js/",
-    "githubusercontent.com/scala/"
+    "githubusercontent.com/scala/",
+    "MacrotaskExecutor.scala" // TODO temporary workaround
```
Comment (armanbilge) [inline core/js/src/main/scala/cats/effect/tracing/TracingPlatform.scala:86]: This _should_ be covered by `githubusercontent.com/scala-js/` except somehow these sourcemap urls are bungled for Macrotask.

Comment (armanbilge) [reply core/js/src/main/scala/cats/effect/tracing/TracingPlatform.scala:86]: The problem is actually in sbt-spiewak: https://github.com/djspiewak/sbt-spiewak/issues/76

**Comment by armanbilge** [comment]
This is actually all good :) I was getting tripped up by the example app seeming to be over-aggressively filtered because it lives inside the CE3 repo 😅

**Comment by armanbilge** [comment]
This is what you get when you run the example app from the tracing docs:
https://typelevel.org/cats-effect/docs/tracing#complete-code

```
The 20th fibonacci number is 17711
java.lang.Throwable: Boom!
  at null.<anonymous>(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:35:55)
  at apply @ <jscode>.null.<anonymous>(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:35)
  at ifM @ <jscode>.null.<anonymous>(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:35)
  at apply @ <jscode>.null.<anonymous>(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:34)
  at flatMap @ <jscode>.null.<anonymous>(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:34)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at apply @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.fib(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:24)
  at flatMap @ Example$.program(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:33)
  at null.$f_Lcats_effect_IOApp @ <jscode>.null.$s_LExample__main__AT__V(/workspace/cats-effect/example/js/src/main/scala/cats/effect/example/Example.scala:21)
```

**Comment by armanbilge** [comment]
For comparison, I also ran that program on the JVM. Very pleased to see a 1-for-1 correspondence :)
```
The 20th fibonacci number is 17711
java.lang.Throwable: Boom!
        at Example$.$anonfun$program$5(Example.scala:35)
        at apply @ Example$.$anonfun$program$3(Example.scala:35)
        at ifM @ Example$.$anonfun$program$3(Example.scala:35)
        at apply @ Example$.$anonfun$program$1(Example.scala:34)
        at flatMap @ Example$.$anonfun$program$1(Example.scala:34)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at apply @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.fib(Example.scala:24)
        at flatMap @ Example$.program(Example.scala:33)
        at run$ @ Example$.run(Example.scala:21)
```

File: core/js/src/main/scala/cats/effect/IOApp.scala (hunk 1 above)
Comment (armanbilge) [reply core/js/src/main/scala/cats/effect/IOApp.scala:228]: For Node.js this works, but for browsers where `reportExitCode` is a no-op maybe we still want to throw it. Or at least throw _something_ :)

Comment (armanbilge) [reply core/js/src/main/scala/cats/effect/IOApp.scala:228]: On browsers, the cancellation case above will also do nothing.

**Review by djspiewak — APPROVED** [review]

**Pull request merged**